### PR TITLE
-completed react-events-and-state

### PR DESCRIPTION
**Pull request by btoomepuu**
https://user-images.githubusercontent.com/59620542/188040157-352cc843-3331-4e6b-8607-0b6d40c17c1b.mov

### CODE REVIEW

File: react-events-and-state/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { isClicked: false };
+  }
+
+  handleClick() {
+    this.setState({ isClicked: true });
+    this.props.text = 'Thank you!';
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx]: `this.props` should be read only, we never directly set `this.props` from inside the component.

File: react-events-and-state/src/index.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  render() {
+    return <button onClick={this.handleClick}>{this.props.text}</button>;
+  }
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx]: Here you should use an `if` statement to set the correct text in the button based on the value in `this.state`, not `this.props`.
```JSX
if (this.state.isClicked) {
  return <button>Thank You!</button>;
} else {
  return <button onClick={this.handleClick}>Click Me!</button>;
}
```

File: react-events-and-state/src/index.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+const container = document.querySelector('#root');
+const root = ReactDOM.createRoot(container);
+
+const element = <CustomButton text="Click Me!" />;
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx]: No need to pass the `text` props here, for this exercise.

**Review by scbowler — CHANGES_REQUESTED** [review]
A few updates needed. Let me know if you have any questions.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**